crates/workers/src/workers.rs

```
@@ -51,24 +72,42 @@ impl Workers {
     pub async fn from_path(
         workers_dir: PathBuf,
         key_storage: Arc<KeyStorage>,
-        scope: Scope,
+        scope: PeerScope,
     ) -> eyre::Result<Self> {
         let workers = load_persisted_workers(workers_dir.as_path()).await?;
         let mut worker_ids = HashMap::with_capacity(workers.len());
         let mut worker_infos = HashMap::with_capacity(workers.len());
+        let mut runtimes = HashMap::with_capacity(workers.len());
 
         for w in workers {
             let worker_id = w.worker_id;
             let deal_id = w.deal_id.clone();
+            let cu_count = w.cu_count;
             worker_infos.insert(worker_id, w.into());
             worker_ids.insert(deal_id, worker_id);
+
+            let runtime = tokio::task::spawn_blocking(move || {
+                let runtime = tokio::runtime::Builder::new_multi_thread()
+                    .thread_name(format!("worker-pool-{}", worker_id))
+                    .worker_threads(cu_count)
+                    .max_blocking_threads(cu_count)
+                    .build()
+                    .map_err(|err| WorkersError::CreateRuntime { err })
+                    .unwrap();
```

panic in the constructor, is it the only way? maybe return an error and bubble it up, so that it finally can be handled in Aqua?

---

No, it was mistake. FIxed